build.ps1

```diff
@@ -1,183 +1,72 @@
-    $requiredVersionOfplatyPS = 0.9
-    if ($null -eq (Get-Module platyPS -ListAvailable -Verbose:$verbosity | Where-Object { $_.Version -ge $requiredVersionOfplatyPS }))
-    {
-        "Cannot find required minimum version $requiredVersionOfplatyPS of platyPS. Please install it from https://www.powershellgallery.com/packages/platyPS/ using e.g. the following command: Install-Module platyPS"
-    }
-    if ($null -eq (Get-Module platyPS -Verbose:$verbosity))
-    {
-        Import-Module platyPS -Verbose:$verbosity
-    }
-    if (-not (Test-Path $markdownDocsPath -Verbose:$verbosity))
-    {
-        throw "Cannot find markdown documentation folder."
-    }
-    New-ExternalHelp -Path $markdownDocsPath -OutputPath $outputDocsPath -Force -Verbose:$verbosity
-}
-
-# Appveyor errors out due to $profile being null. Hence...
-$moduleRootPath = "$HOME/Documents/WindowsPowerShell/Modules"
-if ($null -ne $profile)
-{
-    $moduleRootPath = Join-Path (Split-Path $profile) 'Modules'
-}
-$modulePSSAPath = Join-Path $moduleRootPath 'PSScriptAnalyzer'
-if ($Install)
-{
-    if (-not (Test-Path $moduleRootPath))
-    {
-        New-Item -Path $moduleRootPath -ItemType Directory -Force -Verbose:$verbosity
-    }
-    if (-not (Test-Path -Path $destinationPath))
-    {
-        throw "Please build the module first."
-    }
-    Copy-Item -Path $destinationPath -Destination $modulePSSAPath -Recurse -Verbose:$verbosity
-}
-
-if ($Test)
-{
-    Import-Module -Name Pester -MinimumVersion 4.3.1 -ErrorAction Stop
-    Function GetTestRunnerScriptContent($testPath)
-    {
-        $x = @"
-        cd $testPath
-        Invoke-Pester
-"@
-        return $x
-    }
-
-    Function CreateTestRunnerScript($testPath)
-    {
-        $tmptmpFilePath = [System.IO.Path]::GetTempFileName()
-        $tmpFilePath = $tmptmpFilePath + '.ps1'
-        Move-Item $tmptmpFilePath $tmpFilePath -Verbose:$verbosity
-        $content = GetTestRunnerScriptContent $testPath
-        Set-Content -Path $tmpFilePath -Value $content -Verbose:$verbosity
-        return $tmpFilePath
-    }
-
-    Function GetTestPath($TestType)
-    {
-        if ($TestType -eq "engine")
-        {
-            $testPath = Join-Path $projectRoot "Tests/Engine"
-        }
-        else
-        {
-            $testPath = Join-Path $projectRoot "Tests/Rules"
+END {
+    Import-Module -Force (Join-Path $PSScriptRoot build.psm1)
+    if ( $Clean -or $Clobber ) {
+        Remove-Build
+        if ( $PSCmdlet.ParameterSetName -eq "Clean" ) {
+            return
         }
-        return $testPath
     }
 
-    Function RunTest($TestType, [Boolean] $DifferentProcess)
-    {
-        $testPath = GetTestPath($TestType)
-        if ($DifferentProcess)
-        {
-            $testScriptFilePath = CreateTestRunnerScript $testPath
-            Start-Process powershell -ArgumentList "-NoExit","-File $testScriptFilePath" -Verb runas
-            # clean up the test file
+    $setName = $PSCmdlet.ParameterSetName
```

Comment:
Since this is used above, maybe it's worth assigning above, below the `Import-Module` line?